app/src/main/java/com/cdio/solitaire/model/CardStack.kt

```
@@ -0,0 +1,144 @@
+package com.cdio.solitaire.model
+
+import android.util.Log
+
+// TODO: Consider adding nullable suit var for foundations
+// TODO: Consider adding isEmpty method for code brevity
+
+class CardStack(val stackID: Int) {
+    var head: Card? = null
+    var tail: Card? = null
+    var size: Int = 0
+    var hiddenCards: Int = 0
+
+    /**
+     * Standard push, pushing an element to the tail of the stack
+     */
+    fun pushCard(card: Card) {
+        when (size) {
+            0 -> {
+                head = card
+                tail = card
+                size++
+            }
+            else -> {
+                tail!!.next = card
+                card.prev = tail
+                tail = card
+                size++
+            }
+        }
+        card.stackID = stackID
+    }
+
+    /**
+     * Standard pop, popping and returning the tail element of the stack
+     */
+    fun popCard(): Card? {
+        val poppedCard: Card? = tail
+        when (size) {
+            0 -> {
+                Log.e("EmptyStackPop", "Stack: $stackID")
+                return null
```

Consider throwing an exception here instead of just logging the error, since a stack trace maybe could provide a lot of useful information.

That also means the method doesn't have to return a nullable object, which can make the code simpler.